std/math.d

```diff
@@ -2532,74 +2532,161 @@ unittest
  *      $(TR $(TD $(NAN))            $(TD FP_ILOGBNAN) $(TD no))
  *      )
  */
-int ilogb(real x)  @trusted nothrow @nogc
+int ilogb(T)(const T x) @trusted nothrow @nogc
+    if(isFloatingPoint!T)
 {
-    version (Win64_DMD_InlineAsm)
+    import core.bitop : bsr;
+    // Provide a bsr implementation for ulong in 32-bit mode
+    int bsr_ulong(ulong x) @trusted pure nothrow @nogc
     {
-        asm pure nothrow @nogc
+        static if (size_t.sizeof == 4)
         {
-            naked                       ;
-            fld     real ptr [RCX]      ;
-            fxam                        ;
-            fstsw   AX                  ;
-            and     AH,0x45             ;
-            cmp     AH,0x40             ;
-            jz      Lzeronan            ;
-            cmp     AH,5                ;
-            jz      Linfinity           ;
-            cmp     AH,1                ;
-            jz      Lzeronan            ;
-            fxtract                     ;
-            fstp    ST(0)               ;
-            fistp   dword ptr 8[RSP]    ;
-            mov     EAX,8[RSP]          ;
-            ret                         ;
+            size_t msb = x >> 32;
+            size_t lsb = cast(size_t) x;
+            if (msb)
+                return bsr(msb) + 32;
+            else
+                return bsr(lsb);
+        }
+        else
+        {
+            return bsr(x);
+        }
+    }
 
-        Lzeronan:
-            mov     EAX,0x80000000      ;
-            fstp    ST(0)               ;
-            ret                         ;
+    alias F = floatTraits!T;
 
-        Linfinity:
-            mov     EAX,0x7FFFFFFF      ;
-            fstp    ST(0)               ;
-            ret                         ;
+    Unqual!T vf = x;
+    ushort* vu = cast(ushort*)&vf;
+    uint* vui = cast(uint*)&vf;
+    static if(is(Unqual!T == float))
+        int* vi = cast(int*)&vf;
+    else
+        long* vl = cast(long*)&vf;
+
+    int ex = vu[F.EXPPOS_SHORT] & F.EXPMASK;
+    static if (F.realFormat == RealFormat.ieeeExtended)
+    {
+        if (ex)
+        {   // If exponent is non-zero
```

Comment:
I do believe the coding style of Phobos discourages comments nestled in on the same line as opening braces.  Please move this and others on the next line.